tartufo/scanner.py

```diff
@@ -311,18 +358,40 @@ def scan_entropy(self, chunk: types.Chunk) -> List[Issue]:
                 hex_strings = util.get_strings_of_set(word, HEX_CHARS)
 
                 for string in b64_strings:
-                    if not self.signature_is_excluded(string, chunk.file_path):
-                        b64_entropy = self.calculate_entropy(string, BASE64_CHARS)
-                        if b64_entropy > 4.5:
-                            issues.append(Issue(types.IssueType.Entropy, string, chunk))
+                    self.evaluate_entropy_string(
+                        chunk, issues, string, BASE64_CHARS, 4.5
+                    )
 
                 for string in hex_strings:
-                    if not self.signature_is_excluded(string, chunk.file_path):
-                        hex_entropy = self.calculate_entropy(string, HEX_CHARS)
-                        if hex_entropy > 3:
-                            issues.append(Issue(types.IssueType.Entropy, string, chunk))
+                    self.evaluate_entropy_string(chunk, issues, string, HEX_CHARS, 3)
+
         return issues
 
+    def evaluate_entropy_string(
+        self,
+        chunk: types.Chunk,
+        issues: List[Issue],
+        string: str,
+        chars: str,
+        min_entropy_score: float,
+    ):
+        """
+        Check entropy string using entropy characters and score.
+
+        :param chunk: The chunk of data to check
+        :param issues: Issue list to append any strings flagged
+        :param string: String to check
+        :param chars: Characters to calculate score
+        :param min_entropy_score: Minimum entropy score to flag
+        """
+        if not self.signature_is_excluded(string, chunk.file_path):
+            entropy_score = self.calculate_entropy(string, chars)
+            if entropy_score > min_entropy_score:
+                if self.entropy_string_is_excluded(string, chunk.file_path):
+                    self.logger.debug("entropy string %s was excluded", string)
+                else:
+                    issues.append(Issue(types.IssueType.Entropy, string, chunk))
```

Comment:
I'm a little hesitant on appending to the list inline here, and not returning anything. That's pretty uncommon behavior in Python, and could easily be missed as unexpected behavior. I think it might be best if this were to return the list of issues it found, and then the caller can do
```python
issues += self.evaluate_entropy_string(...)
```
It's just a bit more explicit and obvious what's happening that way.

But, I do love that this logic is split out! Great call on that! Much better for testing and readability!

Reply:
Good idea. Updated

Reply:
cc @tarkatronic